src/timely-util/src/order.rs

```diff
@@ -81,6 +82,32 @@ use timely::PartialOrder;
 #[derive(Debug, Clone, PartialEq, Eq, PartialOrd, Ord, Hash, Serialize, Deserialize)]
 pub struct Partitioned<P, T>(Product<Interval<P>, T>);
 
+impl<P: fmt::Display, T: fmt::Display> fmt::Display for Partitioned<P, T> {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> Result<(), fmt::Error> {
+        f.write_str("(")?;
+        match self.interval() {
+            Interval::Range(lower, upper) => {
+                match lower {
+                    RangeBound::Elem(p) => p.fmt(f)?,
+                    RangeBound::Bottom => f.write_str("-inf")?,
+                    RangeBound::Top => unreachable!(),
+                }
+                f.write_str("..")?;
+                match upper {
+                    RangeBound::Elem(p) => p.fmt(f)?,
+                    RangeBound::Top => f.write_str("+inf")?,
+                    RangeBound::Bottom => unreachable!(),
+                }
+            }
+            Interval::Point(p) => p.fmt(f)?,
+        }
+        f.write_str(", ")?;
+        self.timestamp().fmt(f)?;
+        f.write_str(")")?;
```

Comment:
when its not a range, it wont be easy to tell which of the pair is the ts and which is the Partition? maybe in practice this wont matter

Reply:
how do you mean? this prints partitioned frontiers like this where the partition is always the first element and if it's a range it has the rust range `..` dots

```
source_upper={(-inf..0, 0), (0..1, 0), (1..+inf, 0), (0, 6570), (1, 3430)}
```